multitoken/src/lib.rs

```diff
@@ -0,0 +1,112 @@
+pub struct SimpleMTK {
+    #[MTKStateKeeper]
+    pub tokens: MTKState,
+    pub token_id: TokenId,
+    pub owner: ActorId,
+    pub supply: BTreeMap<TokenId, u128>,
+}
+
+pub trait SimpleMTKCore: MTKCore {
+    fn mint(&mut self, amount: u128, token_metadata: Option<TokenMetadata>);
+
+    fn burn(&mut self, id: TokenId, amount: u128);
+}
+
+static mut CONTRACT: Option<SimpleMTK> = None;
+
+gstd::metadata! {
+    title: "MTK",
+    init:
+        input: InitMTK,
+    handle:
+        input: MyMTKAction,
+        output: Vec<u8>,
+    state:
+        input: MTKQuery,
+        output: MTKQueryReply,
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn init() {
+    let config: InitMTK = msg::load().expect("Unable to decode InitConfig");
+    let mut multi_token = SimpleMTK::default();
+    multi_token.tokens.name = config.name;
+    multi_token.tokens.symbol = config.symbol;
+    multi_token.tokens.base_uri = config.base_uri;
+    multi_token.owner = msg::source();
+    CONTRACT = Some(multi_token);
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn handle() {
+    let action: MyMTKAction = msg::load().expect("Could not load msg");
+    let multi_token = CONTRACT.get_or_insert(SimpleMTK::default());
+    match action {
+        MyMTKAction::Mint {
+            amount,
+            token_metadata,
+        } => SimpleMTKCore::mint(multi_token, amount, token_metadata),
+        MyMTKAction::Burn { id, amount } => SimpleMTKCore::burn(multi_token, id, amount),
+        MyMTKAction::BalanceOf { account, id } => MTKCore::balance_of(multi_token, &account, &id),
+        MyMTKAction::BalanceOfBatch { accounts, ids } => {
+            MTKCore::balance_of_batch(multi_token, &accounts, &ids)
+        }
+        MyMTKAction::MintBatch {
+            ids,
+            amounts,
+            tokens_metadata,
+        } => MTKCore::mint_batch(multi_token, &msg::source(), &ids, &amounts, tokens_metadata),
+        MyMTKAction::TransferFrom {
+            from,
+            to,
+            id,
+            amount,
+        } => MTKCore::transfer_from(multi_token, &from, &to, &id, amount),
+        MyMTKAction::BatchTransferFrom {
+            from,
+            to,
+            ids,
+            amounts,
+        } => MTKCore::batch_transfer_from(multi_token, &from, &to, &ids, &amounts),
+        MyMTKAction::BurnBatch { ids, amounts } => MTKCore::burn_batch(multi_token, &ids, &amounts),
+        MyMTKAction::Approve { account } => MTKCore::approve(multi_token, &account),
+        MyMTKAction::RevokeApproval { account } => MTKCore::revoke_approval(multi_token, &account),
+    }
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn meta_state() -> *mut [i32; 2] {
+    let query: MTKQuery = msg::load().expect("failed to decode input argument");
+    let multi_token = CONTRACT.get_or_insert(SimpleMTK::default());
+    let encoded = MTKTokenState::proc_state(multi_token, query).expect("error");
+    gstd::util::to_leak_ptr(encoded)
+}
+
+impl SimpleMTKCore for SimpleMTK {
```

Comment:
Is it implementation for fungible tokens?

Reply:
Indeed, this implementation just tracks the amount of tokens minted and automatically assigns ids. That's it.